### PR TITLE
Add initial crawl on connect

### PR DESCRIPTION
**Pull request by HardeepAsrani**
### All Submissions:

* [x] Have you followed the [Optimole Contributing guideline](https://github.com/Codeinwp/optimole-wp/blob/master/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change? 
 

### Changes proposed in this Pull Request:
This crawls images when you connect to Optimole service. More details here: https://github.com/Codeinwp/optimole-lambda-functions/pull/388
 
 

Closes https://github.com/Codeinwp/optimole-service/issues/643.

### How to test the changes in this Pull Request:
- Create a new website and add some images to it, you can add 100-150 images for testing. (If you don't want to upload, it's easy to use https://wordpress.org/plugins/instant-images/)
- Once you connect your website to Optimole and the loading stops, you should see some images crawled and ready for display in the dashboard.
- Make sure you're testing on a new instance, not a website which has already been crawled by Optimole.
- It's an old video but the result should be something like this: https://www.loom.com/share/7a32c468673b4b9884cb7cf51c71355d

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: inc/rest.php [hunk 1]
```diff
@@ -423,6 +435,63 @@ public function get_sample_rate( WP_REST_Request $request ) {
 		return $this->response( $image );
 	}
 
+	/**
+	 * Crawl & upload initial load.
+	 *
+	 * @return boolean True if no more posts left to receive, false if there are more left.
```
Comment (GrigoreMihai) [inline inc/rest.php:441]: This response should be `WP_REST_Response` , it's not breaking anything but just to be accurate.

**Review by GrigoreMihai — CHANGES_REQUESTED** [review]
Overall it looks good just those small mentions.

File: inc/api.php [hunk 2]
```diff
@@ -21,6 +21,12 @@ final class Optml_Api {
 	 * @var string Api root.
 	 */
 	private $upload_api_root = 'https://generateurls-prod.i.optimole.com/upload';
+	/**
+	 * Optimole onboard api root url.
+	 *
+	 * @var string Api root.
+	 */
+	private $onboard_api_root = 'https://onboard.ha.optml.cloud/onboard_api/';
```
[the remark this answers is not in the document]
Comment (HardeepAsrani) [reply inc/api.php:29]: Yea, sorry, forgot to replace ha. subdomain.

File: inc/api.php [hunk 3]
```diff
@@ -21,6 +21,12 @@ final class Optml_Api {
 	 * @var string Api root.
 	 */
 	private $upload_api_root = 'https://generateurls-prod.i.optimole.com/upload';
+	/**
+	 * Optimole onboard api root url.
+	 *
+	 * @var string Api root.
+	 */
+	private $onboard_api_root = 'https://onboard.i.optml.cloud/onboard_api/';
```
Comment (GrigoreMihai) [inline inc/api.php:29]: This needs to be `https://onboard.i.optimole.com/onboard_api/` .

**Review by GrigoreMihai — CHANGES_REQUESTED** [review]

**Comment by GrigoreMihai** [comment]
For solving the conflicts I would recommend to first merge https://github.com/Codeinwp/optimole-wp/pull/478 on development and then sync.

**Review by GrigoreMihai — APPROVED** [review]

**Comment by HardeepAsrani** [comment]
Is anything else needed here, do we put it for QA testing or something else? @GrigoreMihai

**Comment by GrigoreMihai** [comment]
> Is anything else needed here, do we put it for QA testing or something else? @GrigoreMihai

We can put it for testing after the staging deploy build finishes https://github.com/Codeinwp/optimole-lambda-functions/actions/runs/4280421992/jobs/7452172229 .

I would also add a bit more details about how to test, but it's just a suggestion.

**Comment by irinelenache** [comment]
@HardeepAsrani @GrigoreMihai I tried to test it but i'm not sure it worked. Am i supposed to test using the API key from https://staging-dashboard.optimole.com/ and the [test plugin](https://github.com/Codeinwp/optimole-test)?

**Comment by GrigoreMihai** [comment]
> @HardeepAsrani @GrigoreMihai I tried to test it but i'm not sure it worked. Am i supposed to test using the API key from https://staging-dashboard.optimole.com/ and the [test plugin](https://github.com/Codeinwp/optimole-test)?

Yup, the staging has the changes required, the testing plugin needed the constant for this API, updated it now, should work with the latest test plugin, sorry for this, I thought it was added by Hardeep there .

**Comment by irinelenache** [comment]
@GrigoreMihai @HardeepAsrani Tested again and now everything works fine regarding this PR, thank you 🚀

**Comment by HardeepAsrani** [comment]
Thank you @irinelenache, I'll fix the merge conflicts.

**Pull request merged**